packages/dds/merge-tree/src/properties.ts

```diff
@@ -96,7 +96,7 @@ export function matchProperties(a: PropertySet | undefined, b: PropertySet | und
 			}
 		}
 	} else {
-		if (b) {
+		if (b && Object.entries(b).length > 0) {
```

Comment:
we should minimize the enumerations as much as possible. i would fist check if a and b are undefined, this should be the most common case, and then we never need to enumerate the objects. not sure what Object.entries returns, but maybe keep it, and resuse in other check, and in place of the for...in, as they internally do a similar thing

Reply:
i consider this code performance critical, so every call, and check we can avoid the better

Reply:
I tried to optimize matchProperties a bit, if you have any suggestions to improve it more please let me know. I definitely think it can be made better still but I'm having some trouble figuring out how.